core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -699,12 +689,15 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
       ((!appCompleted && fastInProgressParsing) || reparseChunkSize > 0)
 
     val bus = new ReplayListenerBus()
-    val listener = new AppListingListener(fileStatus, clock, shouldHalt)
+    val listener = new AppListingListener(reader, clock, shouldHalt)
     bus.addListener(listener)
 
     logInfo(s"Parsing $logPath for listing data...")
-    Utils.tryWithResource(EventLoggingListener.openEventLog(logPath, fs)) { in =>
-      bus.replay(in, logPath.toString, !appCompleted, eventsFilter)
+    val logFiles = reader.listEventLogFiles
+    logFiles.foreach { file =>
+      Utils.tryWithResource(EventLogFileReader.openEventLog(file.getPath, fs)) { in =>
+        bus.replay(in, file.getPath.toString, !appCompleted, eventsFilter)
```

Comment:
For `RollingEventLogFilesFileReader`, for example, if `AppListingListener` throws `HaltReplayException` while replaying file_1, it seems we'll still continue to replay for following files (e.g. file_2, file_3, ....), right ? We should avoid it.

Reply:
Nice finding! Will address.